drivers/i2c/CMakeLists.txt

```diff
@@ -4,80 +4,63 @@ zephyr_syscall_header(${ZEPHYR_BASE}/include/zephyr/drivers/i2c.h)
-zephyr_library_sources_ifdef(CONFIG_I2C_ANDES_ATCIIC100		i2c_andes_atciic100.c)
-zephyr_library_sources_ifdef(CONFIG_I2C_SC18IM704	i2c_sc18im704.c)
-zephyr_library_sources_ifdef(CONFIG_I2C_SMARTBOND	i2c_smartbond.c)
-zephyr_library_sources_ifdef(CONFIG_I2C_XILINX_AXI	i2c_xilinx_axi.c)
-zephyr_library_sources_ifdef(CONFIG_I2C_MCHP_MSS		i2c_mchp_mss.c)
-zephyr_library_sources_ifdef(CONFIG_I2C_SEDI		i2c_sedi.c)
-zephyr_library_sources_ifdef(CONFIG_I2C_AMBIQ		i2c_ambiq.c)
-zephyr_library_sources_ifdef(CONFIG_I2C_ENE_KB1200	i2c_ene_kb1200.c)
-zephyr_library_sources_ifdef(CONFIG_GPIO_I2C_SWITCH	gpio_i2c_switch.c)
-zephyr_library_sources_ifdef(CONFIG_I2C_NUMAKER		i2c_numaker.c)
-
-zephyr_library_sources_ifdef(CONFIG_I2C_STM32_V1
-	i2c_ll_stm32_v1.c
-	i2c_ll_stm32.c
-	)
-zephyr_library_sources_ifdef(CONFIG_I2C_STM32_V2
-	i2c_ll_stm32_v2.c
-	i2c_ll_stm32.c
-	)
+add_subdirectory_ifdef(CONFIG_I2C_TARGET target)
 
-zephyr_library_sources_ifdef(CONFIG_I2C_TEST		i2c_test.c)
+zephyr_library_sources(i2c_common.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_BITBANG								i2c_bitbang.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_EMUL									i2c_emul.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_GPIO									i2c_gpio.c)
+zephyr_library_sources_ifdef(CONFIG_USERSPACE									i2c_handlers.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_RTIO									i2c_rtio.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_SHELL									i2c_shell.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_TEST									i2c_test.c)
 
-zephyr_library_sources_ifdef(CONFIG_USERSPACE		i2c_handlers.c)
+zephyr_library_sources_ifdef(CONFIG_GPIO_I2C_SWITCH						gpio_i2c_switch.c)
 
-add_subdirectory_ifdef(CONFIG_I2C_TARGET target)
+zephyr_library_sources_ifdef(CONFIG_I2C_AMBIQ									i2c_ambiq.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_ANDES_ATCIIC100				i2c_andes_atciic100.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_TELINK_B91						i2c_b91.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_CC13XX_CC26XX					i2c_cc13xx_cc26xx.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_CC32XX								i2c_cc32xx.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_DW										i2c_dw.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_ENE_KB1200						i2c_ene_kb1200.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_ESP32									i2c_esp32.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_GD32									i2c_gd32.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_GECKO									i2c_gecko.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_INFINEON_CAT1					i2c_ifx_cat1.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_INFINEON_XMC4					i2c_ifx_xmc4.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_IMX										i2c_imx.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_ITE_ENHANCE						i2c_ite_enhance.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_ITE_IT8XXX2						i2c_ite_it8xxx2.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_LITEX									i2c_litex.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_STM32									i2c_ll_stm32.c)
+zephyr_library_sources_ifdef(CONFIG_I2C_STM32_V1							i2c_ll_stm32_v1.c)
```

Comment:
i2c_ll_stm32 is the top level driver file with implementation details in the v1 and v2, the older sources_ifdef made a bit more sense here

Reply:
The top level driver Kconfig [CONFIG_I2C_STM32](https://docs.zephyrproject.org/latest/kconfig.html#CONFIG_I2C_STM32) will be enabled when either `CONFIG_I2C_STM32_V1` / `CONFIG_I2C_STM32_V2` is enabled, the driver file is compiled in the line above (L36)